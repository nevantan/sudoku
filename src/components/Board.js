import React from 'react';

const Board = () => <p>Board Component</p>;

export default Board;
