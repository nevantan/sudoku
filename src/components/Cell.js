import React from 'react';
import { connect } from 'react-redux';
import { updateCell } from '../actions/board';
import { setActive } from '../actions/settings';

export class Cell extends React.Component {
  constructor(props) {
    super(props);
  }
  handleChange = e => {
    const value = e.target.value.replace(this.props.value, '');
    if (value.match(/[0-9]/)) {
      this.props.updateCell(this.props.row, this.props.column, parseInt(value));
      this.props.setActive(parseInt(value));
    }
  };
  handleKeyDown = e => {
    if (e.key == 'Backspace' && !this.props.fixed) {
      this.props.updateCell(this.props.row, this.props.column, 0);
    } else if (e.key == 'ArrowRight' && this.props.column < 8) {
      document
        .getElementById('c' + this.props.row + (this.props.column + 1))
        .focus();
    } else if (e.key == 'ArrowLeft' && this.props.column > 0) {
      document
        .getElementById('c' + this.props.row + (this.props.column - 1))
        .focus();
    } else if (e.key == 'ArrowUp' && this.props.row > 0) {
      document
        .getElementById('c' + (this.props.row - 1) + this.props.column)
        .focus();
    } else if (e.key == 'ArrowDown' && this.props.row < 8) {
      document
        .getElementById('c' + (this.props.row + 1) + this.props.column)
        .focus();
    }
  };
  handleFocus = e => {
    this.props.setActive(this.props.value);
  };
  render() {
    let classes = 'cell';
    if (!this.props.valid) classes += ' error';
    if (this.props.fixed) classes += ' fixed';
    if (this.props.active) classes += ' highlight';

    return (
      <input
        id={'c' + this.props.id}
        className={classes}
        type="text"
        value={this.props.value ? this.props.value : ''}
        onChange={!this.props.fixed ? this.handleChange : undefined}
        onKeyDown={this.handleKeyDown}
        onFocus={this.handleFocus}
        readOnly={this.props.fixed}
      />
    );
  }
}

const mapStateToProps = (state, { row, column }) => {
  const cell = state.board[row][column];
  return {
    value: cell.value,
    valid: cell.valid,
    fixed: cell.fixed,
    active: state.settings.activeValue === cell.value && cell.value !== 0
  };
};

const mapDispatchToProps = dispatch => ({
  updateCell: (row, column, value) => dispatch(updateCell(row, column, value)),
  setActive: value => dispatch(setActive(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(Cell);
export { mapStateToProps, mapDispatchToProps };
