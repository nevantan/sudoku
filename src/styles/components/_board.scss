.board {
  display: inline-block;
  border: 1px solid $dark-grey;
}

.board__row {
  display: flex;
}
