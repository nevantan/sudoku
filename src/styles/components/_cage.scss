.cage {
  border: 1px solid $dark-grey;
}
