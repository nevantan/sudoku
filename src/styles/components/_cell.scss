.cell {
  border: 1px solid $light-grey;
  font-size: 3.2rem;
  height: 50px;
  outline: none;
  padding: 1rem;
  text-align: center;
  width: 50px;
  color: #777;
}

.cell.error {
  color: #cc0000;
}

.cell:focus {
  border: 2px solid $highlight;
}

.cell.fixed {
  color: #000;
}

.cell.fixed.error {
  color: #880000;
}
