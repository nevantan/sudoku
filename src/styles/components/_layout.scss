header {
  text-align: center;
  width: 100%;
}

header h1 {
  font-family: sans-serif;
  font-size: 4.8rem;
}

.wrapper {
  display: flex;
  justify-content: space-around;
  margin: 0 auto;
  width: 60rem;
}
