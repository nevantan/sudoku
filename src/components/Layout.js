import React from 'react';
import Board from './Board';

const Layout = () => (
  <div>
    <header>
      <h1>Sudoku</h1>
    </header>
    <div className="wrapper">
      <Board />
    </div>
  </div>
);

export default Layout;
