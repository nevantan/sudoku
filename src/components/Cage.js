import React from 'react';
import Cell from './Cell';

class Cage extends React.Component {
  constructor(props) {
    super(props);
  }
  componentWillMount = () => {
    const cage = this.props.cage;
    this.row = Math.floor(cage / 3) * 3;
    this.col = (cage - this.row) * 3;
  };
  render() {
    return (
      <div>
        <p>Cage Component</p>
        <Cell row={this.row} column={this.col} />
        <Cell row={this.row} column={this.col + 1} />
        <Cell row={this.row} column={this.col + 2} />
        <Cell row={this.row + 1} column={this.col} />
        <Cell row={this.row + 1} column={this.col + 1} />
        <Cell row={this.row + 1} column={this.col + 2} />
        <Cell row={this.row + 2} column={this.col} />
        <Cell row={this.row + 2} column={this.col + 1} />
        <Cell row={this.row + 2} column={this.col + 2} />
      </div>
    );
  }
}

export default Cage;
