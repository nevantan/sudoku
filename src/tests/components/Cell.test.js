import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, Cell } from '../../components/Cell';

test('should render Cell component correctly', () => {
  const wrapper = shallow(<Cell row={0} column={0} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly map state to props', () => {
  const state = {
    board: [[{ value: 0, valid: true }]]
  };
  const ownProps = { row: 0, column: 0 };
  const props = mapStateToProps(state, ownProps);
  expect(props).toEqual({
    value: 0,
    valid: true
  });
});
