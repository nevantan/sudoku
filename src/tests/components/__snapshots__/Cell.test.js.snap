// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Cell component correctly 1`] = `
<input
  className="cell fixed"
  id="c00"
  onFocus={[Function]}
  onKeyDown={[Function]}
  readOnly={true}
  type="text"
  value={1}
/>
`;

exports[`should render Cell component with active value correctly 1`] = `
<input
  className="cell fixed highlight"
  id="c00"
  onFocus={[Function]}
  onKeyDown={[Function]}
  readOnly={true}
  type="text"
  value={2}
/>
`;

exports[`should render Cell component with preset value correctly 1`] = `
<input
  className="cell fixed"
  id="c00"
  onFocus={[Function]}
  onKeyDown={[Function]}
  readOnly={true}
  type="text"
  value={2}
/>
`;

exports[`should render Cell component with value 0 correctly 1`] = `
<input
  className="cell"
  id="c00"
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  readOnly={false}
  type="text"
  value=""
/>
`;

exports[`should render error Cell component correctly 1`] = `
<input
  className="cell error fixed"
  id="c00"
  onFocus={[Function]}
  onKeyDown={[Function]}
  readOnly={true}
  type="text"
  value={1}
/>
`;
